From 7f94fb8bd2f802cc3171b90f0d8dc9f29deb3176 Mon Sep 17 00:00:00 2001
From: Alexander Pohl <pohl199885@gmail.com>
Date: Mon, 7 Jan 2019 21:23:30 +0100
Subject: [PATCH 2/2] Huaweis media stack doesn't handle intra-refresh-mode, so
 skip the error instead.

Thanks to Chris Vandomelen for pointing that out.

Change-Id: I968bc38b3cfb492af7f658f51de6157d050c76af
---
 media/libstagefright/ACodec.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index def8ada3c..fea288ed8 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -4244,9 +4244,8 @@ status_t ACodec::setupAVCEncoderParameters(const sp<AMessage> &msg) {
     if (msg->findInt32("intra-refresh-mode", &intraRefreshMode)) {
         err = setCyclicIntraMacroblockRefresh(msg, intraRefreshMode);
         if (err != OK) {
-            ALOGE("Setting intra macroblock refresh mode (%d) failed: 0x%x",
-                    err, intraRefreshMode);
-            return err;
+            ALOGE("setupAVCEncoderParameters(): set intra-refresh-mode failed, ignoring..");
+            //return err;
         }
     }
 
-- 
2.19.1

